src/PathautoPatternListBuilder.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/**
+ * @file
+ * Contains Drupal\pathauto\PathautoPatternListBuilder.
+ */
+
+namespace Drupal\pathauto;
+
+use Drupal\Core\Config\Entity\ConfigEntityListBuilder;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Url;
+
+/**
+ * Provides a listing of Pathauto pattern entities.
+ */
+class PathautoPatternListBuilder extends ConfigEntityListBuilder {
```

Comment:
You can probably use the DraggableListBuilder, make sure to set the weight entity key.